tracer/src/Datadog.Trace/Debugger/ProbeStatuses/ProbeStatusPoller.cs

```diff
@@ -82,7 +88,11 @@ private void PausePollerTimer()
         {
             try
             {
-                _pollerTimer?.Change(Timeout.Infinite, Timeout.Infinite);
+                var timer = _pollerTimer;
+                if (timer != null && !_isDisposed)
+                {
+                    timer.Change(Timeout.Infinite, Timeout.Infinite);
```

Comment:
If another thread diposes the timer right before calling Change() and after the if checks, there could still be an issue, right? In this method, I think that you can either use the lock before checking _isDisposed or capture ObjectDisposedException without an error message.